### PR TITLE
[#5881] test(authorization): Support Hive test container with SQL based auhtorization

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?

Support Hive test container with SQL based auhtorization 

### Why are the changes needed?

Fix: #5881 

### Does this PR introduce _any_ user-facing change?

Yes

### How was this patch tested?

Just a test code.

### CODE REVIEW

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java [hunk 1]
```diff
@@ -65,6 +65,7 @@ public class ContainerSuite implements Closeable {
   private static volatile KafkaContainer kafkaContainer;
   private static volatile DorisContainer dorisContainer;
   private static volatile HiveContainer kerberosHiveContainer;
+  private static volatile HiveContainer jdbcHiveContainer;
```
Comment (xunliu) [inline integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:68]: I think it better to change `jdbcHiveContainer` to `sqlBaseAuthHiveContiner`?

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java [hunk 2]
```diff
@@ -231,6 +232,34 @@ public void startKerberosHiveContainer() {
     }
   }
 
+  public void startJdbcHiveContainer(Map<String, String> envVars) {
```
Comment (xunliu) [inline integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:235]: I think it better to change `startJdbcHiveContainer()` to `startSqlBaseAuthHiveContiner()`?

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java [hunk 3]
```diff
@@ -231,6 +232,34 @@ public void startKerberosHiveContainer() {
...
+    // If you want to enable SQL based authorization, you need both set the
+    // `ENABLE_JDBC_AUTHORIZATION` environment.
+    if (envVars == null
+        || (!Objects.equals(envVars.get(HiveContainer.HIVE_RUNTIME_VERSION), HiveContainer.HIVE3))
+        || (!envVars.containsKey(HiveContainer.ENABLE_AUTHORIZATION))) {
```
Comment (xunliu) [inline integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:240]: I think it better to change `HiveContainer.ENABLE_AUTHORIZATION` to `HiveContainer.ENABLE_SQL_BASE_AUTHORIZATION`?

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java [hunk 4]
```diff
@@ -509,6 +538,10 @@ public HiveContainer getKerberosHiveContainer() {
     return kerberosHiveContainer;
   }
 
+  public HiveContainer getJdbcHiveContainer() {
```
Comment (xunliu) [inline integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:541]: I think it better change `getJdbcHiveContainer()` to `getSqlBaseAuthHiveContainer()`?

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/HiveContainer.java [hunk 5]
```diff
@@ -49,6 +49,7 @@ public class HiveContainer extends BaseContainer {
   public static final String HIVE_RUNTIME_VERSION = "HIVE_RUNTIME_VERSION";
   public static final String HIVE2 = "hive2"; // The Hive container default version
   public static final String HIVE3 = "hive3";
+  public static final String ENABLE_AUTHORIZATION = "ENABLE_JDBC_AUTHORIZATION";
```
Comment (xunliu) [inline integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/HiveContainer.java:52]: I think it better to change `ENABLE_AUTHORIZATION` to `ENABLE_SQL_BASE_AUTHORIZATION`?

**Review by xunliu — COMMENTED** [review]
I think better change it `ENABLE_JDBC_AUTHORIZATION` to `ENABLE_SQL_BASE_AUTHORIZATION` in the https://github.com/apache/gravitino/blob/main/dev/docker/hive/start.sh#L43C13-L43C38

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/HiveContainer.java (hunk 5 above)
Comment (jerqi) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/HiveContainer.java:52]: OK, I can create a new Hive image.

Comment (jerqi) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/HiveContainer.java:52]: OK. I will change this.

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java (hunk 1 above)
Comment (jerqi) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:68]: Renamed.

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java (hunk 4 above)
Comment (jerqi) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:541]: Renamed.

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java (hunk 3 above)
Comment (jerqi) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:240]: Renamed.

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java (hunk 2 above)
Comment (jerqi) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:235]: Renamed.

**Review by xunliu — APPROVED** [review]
LGTM

**Pull request merged**